### PR TITLE
feat: Add support for asciidoctor-pdf additional arguments.

### LINKED ISSUE #208: Specifying theme file for PDF export
## Description

At the moment there seems to be no way to specify theme file for the PDF export. I would like to specify one to override the silly table stripes.

## Version info

- Atom version: 1.12.7
- OS: Debian GNU/Linux (Sid)
- `asciidoc-preview` version: 2.7.1
- `language-asciidoc` version: 1.6.2

### PR DESCRIPTION
**Pull request by ldez**
## Description

Add support for asciidoctor-pdf additional arguments.

Useful for themes.

## Screenshots

capture du 2017-02-27 23-05-37

capture du 2017-02-27 23-06-00

Fixes #208

### CODE REVIEW

**Comment by mojavelinux** [comment]
Very cool.

I wonder if we should have a dedicated box for the pdf-style name (and maybe pdf-stylesdir and pdf-fontsdir). Doing so would abstract away the low-level details of setting these.

**Comment by ldez** [comment]
Added more options.

capture du 2017-03-09 21-33-50

**Comment by ldez** [comment]
@mojavelinux are you agree my changes?

**Comment by mojavelinux** [comment]
I'll review shortly.

**Comment by mojavelinux** [comment]
This looks great.

Somewhat unrelated to this change, I have to admit that the box "Use asciidoctor-pdf" is confusing. My first thought when I saw that box is "Does it use something else if I don't check this box?" Instead, I think the box should read:

```
[x] Enable PDF export using Asciidoctor PDF
The asciidoctor-pdf command must be available on your PATH.
```

That makes it clear that the box both enables PDF export and communicates that it will be using an external application.

**Comment by mojavelinux** [comment]
I think we should only link to the theming guide once. Perhaps on the style option.

File: package.json [hunk 1]
```diff
@@ -161,12 +161,42 @@
...
+        },
+        "pdfStyle": {
+          "title": "Style (-a pdf-style=)",
+          "description": "The name of the YAML theme file to load. <br> https://github.com/asciidoctor/asciidoctor-pdf/blob/master/docs/theming-guide.adoc#applying-your-theme",
+          "type": "string",
+          "default": "",
+          "order": 3
+        },
+        "pdfStylesDir": {
+          "title": "Fonts directory (-a pdf-stylesdir=)",
```
Comment (mojavelinux) [inline package.json]: The title of this option should be "Styles directory".

File: package.json [hunk 2]
```diff
@@ -161,12 +161,42 @@
...
+        "pdfFontsDir": {
+          "title": "Fonts directory (-a pdf-fontsdir=)",
+          "description": "The directory where the fonts used by your theme, if any, are located. _Specifying an absolute path is recommended._ <br> https://github.com/asciidoctor/asciidoctor-pdf/blob/master/docs/theming-guide.adoc#applying-your-theme",
+          "type": "string",
+          "default": "",
+          "order": 5
+        },
+        "arguments": {
+          "title": "Command line additional arguments",
+          "description": "ex: `-a pdf-stylesdir=~/resources/themes -a pdf-style=basic`",
```
Comment (mojavelinux) [inline package.json]: Now that we have dedicated options for pdf-stylesdir and pdf-style, the example here should be something like "-a pdf-page-size=Letter".

**Comment by ldez** [comment]
capture du 2017-03-12 13-24-54

File: package.json [hunk 3]
```diff
@@ -165,15 +165,45 @@
       "collapsed": true,
       "properties": {
         "enabled": {
-          "title": "Use asciidoctor-pdf",
-          "description": "[asciidoctor-pdf](https://github.com/asciidoctor/asciidoctor-pdf) must be installed in your system.",
+          "title": "Enable PDF export using Asciidoctor PDF.",
+          "description": "The [asciidoctor-pdf](https://github.com/asciidoctor/asciidoctor-pdf) command must be available on your `PATH`.",
```
Comment (Mogztter) [inline package.json]: "on your `PATH`" ? I've always written "in your `PATH`" :neutral_face:

**Comment by Mogztter** [comment]
LGTM :+1:

File: package.json (hunk 3 above)
Comment (mojavelinux) [inline package.json]: It seems it can go either way. But "in your PATH" seems to be used more frequently.

**Comment by mojavelinux** [comment]
Fantastique !

**Pull request merged**